cloudinit/net/network_manager.py

```diff
@@ -173,18 +173,44 @@ def _set_ip_method(self, family, subnet_type):
         self.config[family]["method"] = method
         self._set_default(family, "may-fail", "false")
 
-    def _add_numbered(self, section, key_prefix, value):
-        """
-        Adds a numbered property, such as address<n> or route<n>, ensuring
-        the appropriate value gets used for <n>.
-        """
+    def _get_next_numbered_section(self, section, key_prefix) -> str:
         if not self.config.has_section(section):
             self.config[section] = {}
         for index in itertools.count(1):
             key = f"{key_prefix}{index}"
             if not self.config.has_option(section, key):
-                self.config[section][key] = value
-                break
+                return key
+        return "not_possible"  # for typing
+
+    def _add_numbered(self, section, key_prefix, value):
+        """
+        Adds a numbered property, such as address<n> or route<n>, ensuring
+        the appropriate value gets used for <n>.
+        """
+        key = self._get_next_numbered_section(section, key_prefix)
+        self.config[section][key] = value
+
+    def _add_route_options(self, section, route, key, value):
+        """Add route options to a given route
+
+        Example:
+        Given:
+          section: ipv4
+          route: route0
+          key: mtu
+          value: 500
+
+        Create line under [ipv4] section:
+            route0_options=mtu=500
+
+        If the line already exists, then append the new key/value pair
+        """
+        if not self.config[section].get(f"{route}_options"):
```

Comment:
We re-create the same string and reference the same dict location multiple times here - and also use nested f strings - a bit messy to grok. Assigning a couple of variables would make this a little easier to read.

compare this:
```python
        if not self.config[section].get(f"{route}_options"):
            self.config[section][f"{route}_options"] = f"{key}={value}"
        else:
            self.config[section][
                f"{route}_options"
            ] = f'{self.config[section][f"{route}_options"]},{key}={value}'
```

with this:

```python
        option_number = f"{route}_options"
        route_options = self.config[section].get(option_number)
        if not route_options:
            self.config[section][option_number] = f"{key}={value}"
        else:
            self.config[section][
                option_number
            ] = f'{route_options},{key}={value}'
```